adk-examples/Python/adk-connector.py

```
@@ -0,0 +1,294 @@
+from falkonryclient import client as Falkonry
+from falkonryclient import schemas as Schemas
+from fileAdapter import FileAdapter
+from multiprocessing import Process
+import pandas as pd
+import io
+import os
+import json
+import time as timepkg
+import logging
+
+# Logging configuration
+logging.basicConfig(
+    level=logging.DEBUG,
+    format="%(asctime)s:%(levelname)s:%(message)s"
+    )
+
+
+
+
+class ADKconn:
+    url = "https://example.falkonry.ai"  ### Your account url
+    token = "token"                      ### Your account token
+    falkonry   = Falkonry(url, token)
+    datastream = Schemas.Datastream()
+    datasource = Schemas.Datasource()
+    field = Schemas.Field()
+    time = Schemas.Time()
+    signal = Schemas.Signal()
+    datastreamId = None
+
+
+    def checkDataIngestion(self, tracker):
+        tracker_obj = None
+        for i in range(0, 12):
+            tracker_obj = self.falkonry.get_status(tracker['__$id'])
+            logging.info(tracker_obj['status'])
+            try:
+                if tracker_obj['status'] == 'FAILED' or tracker_obj['status'] == 'ERROR':
```

Here, we should raise an exception for a retry strategy. Max 3 retries.

---

So after 3 'FAILED' or 'ERROR' messages, I should raise an exception, right? Currently, I am running the loop for 12 times.

---

We should handle different tracker responses differently:
For ERROR we retry for a given set/ configurable limit of errors (we throw the exception after the limit is reached)
For PENDING we wait for a set/ configurable time before retrying